### PR TITLE
Adding input validations for spatial coordinates.

### PR DESCRIPTION
**Pull request by sreekanth-cb**

### CODE REVIEW

File: geo/parse.go [hunk 1]
```diff
@@ -272,29 +275,38 @@ func extract4DCoordinates(thing interface{}) (rv [][][][]float64) {
...
 	var coordValue interface{}
-	var typ string
+
 	if thingVal.Kind() == reflect.Map {
 		iter := thingVal.MapRange()
 		for iter.Next() {
 			if iter.Key().String() == "type" {
-				typ = iter.Value().Interface().(string)
-				// make the shape type case insensitive.
-				typ = strings.ToLower(typ)
```
Comment (abhinavdangeti) [inline geo/parse.go:289]: Would we be giving up on the case insensitivity here ?

File: geo/parse.go [hunk 2]
```diff
@@ -383,23 +395,68 @@ func ExtractGeoShapeCoordinates(coordValue interface{},
...
+		// ignore the contents with invalid entry.
+		if len(coords) == 0 {
+			return nil, typ, false
+		}
+
+		if len(coords) > 2 && typ == EnvelopeType {
+			return nil, typ, false
+		}
+
+		if len(coords) < 2 && typ != MultiPointType {
```
Comment (abhinavdangeti) [inline geo/parse.go:422]: `len(coords) < 2 && typ == MultiPointType` ?

File: geo/parse.go (hunk 1 above)
Comment (sreekanth-cb) [reply geo/parse.go:289]: Nope, it is handled in the new function in parse.go line no:301.

Comment (abhinavdangeti) [reply geo/parse.go:289]: Ack.

**Review by sreekanth-cb — COMMENTED** [review]
.

File: geo/parse.go [hunk 3]
```diff
@@ -383,23 +395,68 @@ func ExtractGeoShapeCoordinates(coordValue interface{},
...
 		typ == EnvelopeType {
-		rv = [][][][]float64{{extract2DCoordinates(coordValue)}}
+		coords := extract2DCoordinates(coordValue)
+
+		// ignore the contents with invalid entry.
+		if len(coords) == 0 {
+			return nil, typ, false
+		}
+
+		if len(coords) > 2 && typ == EnvelopeType {
```
Comment (abhinavdangeti) [inline geo/parse.go:418]: len(coords) != 2 && typ == EnvelopeType

File: geo/parse.go (hunk 2 above)
Comment (abhinavdangeti) [reply geo/parse.go:422]: Ah in that case - I recommend updating the earlier check too :)

Comment (abhinavdangeti) [reply geo/parse.go:422]: This is to avoid 2 unnecessary checks for EnvelopeType when len(coords) < 2

**Review by abhinavdangeti — CHANGES_REQUESTED** [review]

File: geo/parse.go (hunk 3 above)
Comment (abhinavdangeti) [reply geo/parse.go:418]: type == EnvelopeType && len(coords) != 2

File: geo/parse.go (hunk 2 above)
Comment (sreekanth-cb) [reply geo/parse.go:422]: I realize the readability problem here. :)

**Review by abhinavdangeti — APPROVED** [review]
Cheers!

**Pull request merged**